### PR TITLE
Supplying ssh keys and env variable to container for e2e testing

### PR DESCRIPTION
**Pull request by shuklanirdesh82**
This change enables our e2e testing to invoke automated steps from test_run container.

Affected files:

1. build.sh: 

- supplying esx and docker host vm IP Addrs
- passing SSH keys to container

2. vmdk_plugin/Makefile: 

- Modifying ```testasroot``` to supply inputs 
- adding -v to see verbose log for running go tests with an output

Testing Done: Yes

using sample testcase:
```
nshukla-mbp:docker-volume-vsphere nshukla$ cat vmdk_plugin/E2E_Tests/sample_e2e_test.go 
package e2e_test

import "testing"
import "os/exec"
import "fmt"
import "os"

// This is a sample testcase and will be removed after finishing the review.

func TestSomething(t *testing.T) {
	var err error
	var out []byte

	out, err = exec.Command("/usr/bin/ssh", "-o StrictHostKeyChecking no", "root@"+os.Getenv("ESX"), "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", "ls").CombinedOutput()
	fmt.Printf("err=>%s.....out=>%s", err, out)

	out, err = exec.Command("/usr/bin/ssh", "-o StrictHostKeyChecking no", "root@"+os.Getenv("VM1"), "ifconfig").CombinedOutput()
	fmt.Printf("err=>%s.....out=>%s", err, out)
}
```

testing done locally and here is the outcome:
```
../misc/scripts/build.sh testasroot "10.20.105.97" "10.20.105.77" "10.20.105.89"
make: Entering directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'

=> Running target testasroot Thu Jan 5 19:13:42 UTC 2017

GO15VENDOREXPERIMENT=1 go test github.com/vmware/docker-volume-vsphere/vmdk_plugin/drivers/vmdk/vmdkops -cover -v
=== RUN   TestCommands
Detaching loopback device /dev/loop1005
Detaching loopback device /dev/loop1005
Detaching loopback device /dev/loop1005
--- PASS: TestCommands (1.59s)
	cmd_test.go:30: 
		Creating Test Volume with name = [DockerTestVol]...
PASS
coverage: 46.7% of statements
ok  	github.com/vmware/docker-volume-vsphere/vmdk_plugin/drivers/vmdk/vmdkops	1.596s
GO15VENDOREXPERIMENT=1 go test github.com/vmware/docker-volume-vsphere/vmdk_plugin/utils/config -cover -v
=== RUN   TestLoad
--- PASS: TestLoad (0.00s)
PASS
coverage: 64.3% of statements
ok  	github.com/vmware/docker-volume-vsphere/vmdk_plugin/utils/config	0.009s
GO15VENDOREXPERIMENT=1 go test github.com/vmware/docker-volume-vsphere/vmdk_plugin/E2E_Tests -v
=== RUN   TestSomething
```

/CC @ashahi1 @tusharnt

### CODE REVIEW

**Comment by shuklanirdesh82** [comment]
1. creating a separate namespace where all E2E tests should be added (```vmdk_plugin/E2E_Tests/```).

2. ```sample_e2e_test.go``` will be replaced with the actual E2E tests ( adding to make CI pass)

File: vmdk_plugin/Makefile [hunk 1]
```diff
@@ -294,13 +294,14 @@ deploy: deploy-all
 .PHONY: test
 test:
 	$(log_target)
-	$(SCRIPTS)/build.sh testasroot
+	$(SCRIPTS)/build.sh testasroot "$(ESX)" "$(VM1)" "$(VM2)"
 
 .PHONY: testasroot
 testasroot:
```
Comment (kerneltime) [inline vmdk_plugin/Makefile]: We should create a new target e2e-test

Comment (shuklanirdesh82) [inline vmdk_plugin/Makefile]: Thanks @kerneltime ! 

Added a new target and here is the output for your reference:

```
nshukla-mbp:docker-volume-vsphere nshukla$ make e2e-tests
/Library/Developer/CommandLineTools/usr/bin/make --directory=vmdk_plugin e2e-tests

=> Running target e2e-tests Mon Jan 9 11:33:28 PST 2017

../misc/scripts/build.sh e2e-dkrVolDriver-test "10.20.105.97" "10.20.105.77" "10.20.105.89"
make: Entering directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'

=> Running target e2e-dkrVolDriver-test Mon Jan 9 19:33:28 UTC 2017

GO15VENDOREXPERIMENT=1 go test github.com/vmware/docker-volume-vsphere/vmdk_plugin/E2E_Tests -cover -v
=== RUN   TestAccessOption
```

**Comment by msterin** [comment]
LGTM - assuming  I can still run 'make test' as before :-)

**Comment by kerneltime** [comment]
* You need to add the test here as well https://github.com/vmware/docker-volume-vsphere/blob/master/misc/drone-scripts/deploy-and-test-wrapper.sh#L89
* For usability define e2e-test that runs e2e-tests, I can see developers choosing to run e2e-test and forget the extra s.

**Comment by shuklanirdesh82** [comment]
Thanks @msterin & @kerneltime  for the review!

> assuming I can still run 'make test' as before
Yeah, it is, moreover you can invoke e2e-test using ```make e2e-test```

> You need to add the test here as well https://github.com/vmware/docker-volume-vsphere/blob/master/misc/drone-scripts/deploy-and-test-wrapper.sh#L89

Good catch! I have added the new target as well addressed usability comment too. Please have a look at the latest change.

**Pull request merged**